src/components/Popup/Popup.ts

```diff
@@ -0,0 +1,145 @@
+import { nostojs } from "@nosto/nosto-js"
+import { customElement } from "../decorators"
+import { NostoElement } from "../Element"
+import { popupStyles } from "./styles"
+import { assertRequired } from "@/utils/assertRequired"
+
+/**
+ * A custom element that displays popup content with dialog and ribbon slots.
+ * Supports conditional activation based on Nosto segments and persistent closure state.
+ *
+ * @property {string} name - Required name used for analytics and localStorage persistence. The popup's closed state will be remembered.
+ * @property {string} [segment] - Optional Nosto segment that acts as a precondition for activation. Only users in this segment will see the popup.
+ *
+ * @example
+ * Basic popup with dialog and ribbon content:
+ * ```html
+ * <nosto-popup name="promo-popup" segment="5b71f1500000000000000006">
+ *   <div slot="default">
+ *     <h2>Special Offer!</h2>
+ *     <p>Get 20% off your order today</p>
+ *     <button n-close>Close</button>
+ *   </div>
+ *   <div slot="ribbon">
+ *     <span>Limited time!</span>
+ *   </div>
+ * </nosto-popup>
+ * ```
+ */
+@customElement("nosto-popup")
+export class Popup extends NostoElement {
+  /** @private */
+  static attributes = {
+    name: String,
+    segment: String
+  }
+
+  name!: string
+  segment?: string
+
+  constructor() {
+    super()
+    this.attachShadow({ mode: "open" })
+  }
+
+  async connectedCallback() {
+    assertRequired(this, "name")
+
+    if (!(await isPopupShown(this))) {
+      this.style.display = "none"
+      return
+    }
+
+    const state = getPopupState(this.name)
+    const mode = state === "ribbon" ? "ribbon" : "open"
+    renderShadowContent(this, mode)
+    this.addEventListener("click", this.handleClick)
+  }
+
+  private handleClick = (event: Event) => {
+    const target = event.target as HTMLElement
+    const closeElement = target?.closest("[n-close]")
+    const ribbonElement = target?.closest("[n-ribbon]")
+
+    if (closeElement) {
+      event.preventDefault()
+      event.stopPropagation()
+      closePopup(this)
+    } else if (ribbonElement) {
+      event.preventDefault()
+      event.stopPropagation()
+      setPopupToRibbon(this.name)
+      renderShadowContent(this, "ribbon")
+    }
+  }
+}
+
+function renderShadowContent(element: Popup, mode: "open" | "ribbon" = "open") {
+  if (!element.shadowRoot) return
+
+  const dialogHidden = mode === "ribbon" ? "hidden" : ""
+  const ribbonHidden = mode === "open" ? "hidden" : ""
+
+  element.shadowRoot.innerHTML = `
+    <style>${popupStyles}</style>
+    <dialog open part="dialog" class="${dialogHidden}">
+      <slot name="default"></slot>
+    </dialog>
+    <div class="ribbon ${ribbonHidden}" part="ribbon">
+      <slot name="ribbon"></slot>
+    </div>
+  `
+}
+
+function closePopup(element: Popup) {
+  if (element.name) {
+    setPopupClosed(element.name)
+  }
+  element.style.display = "none"
+}
+
+async function isPopupShown(element: Popup): Promise<boolean> {
```

Comment:
remove this function and include segment check in getPopupState()
the segment check should be done as the first step
when the segment check fails "closed" is returned